src/wavpack_local.h

```diff
@@ -20,6 +20,12 @@
 #define FASTCALL
 #endif
 
+#if defined(__WATCOMC__) && defined(OPT_ASM_X86)
```

Comment:
I guess this could be generalized and `__WATCOMC__` condition can be dropped,
but I didn't want to poke the bear..

Reply:
Actually I was thinking that myself, but `_cdecl` is ignored on ARM because it doesn't apply, and sometimes that can generate an error for "ignored attributes". Let me experiment.

Reply:
With ARM, OPT_ASM_X86 should not be defined either..

Reply:
Oh, I didn't read carefully. I was thinking that we could not add a macro at all and simply use `__cdecl` because it's ignored when not x86. While cleaner in some ways, that's liable to generate warnings sometimes. But yes, `__WATCOMC__` can certainly go.

Reply:
>  But yes, `__WATCOMC__` can certainly go.

Well, it still is a problem: only MSVC (and Watcom) support that keyword.
MinGW (gcc targeting windows) does define it as `__attribute__((__cdecl__))`
so that one is OK too. But darwin-targeting gcc or clang doesn't define a
`__cdecl` macro. 

So I suggest that I keep  `__WATCOMC__` and really don't poke the bear:
what do you say?

Reply:
Yeah, makes sense.